### PR TITLE
Fix zero exponent for `int`s

### PR DESCRIPTION
**Pull request by Molkree**
It is typed as `int` instead of `Any` after this PR.

Followup to https://github.com/python/typeshed/pull/6325#discussion_r797710621

### CODE REVIEW

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -176,7 +176,7 @@ class super:
     @overload
     def __init__(self) -> None: ...
 
-_PositiveInteger = Literal[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]
+_PositiveInteger = Literal[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]
```
[the remark this answers is not in the document]
Comment (Molkree) [reply stdlib/builtins.pyi:179]: Sure, I can add another overload instead.

So I guess it should look like this:
```py
    @overload
    def __pow__(self, __x: Literal[0], __modulo: None = ...) -> Literal[1, -1]: ...
```
Not sure if it's important but where would it go, before/after `_PositiveInteger` overload or somewhere else?

P.S. Is it okay to just force push this PR?

[the remark this answers is not in the document]
Comment (Akuli) [reply stdlib/builtins.pyi:179]: `-10 ** 0` is confusing. It actually evalutes `-(10 ** 0)`, not `(-10) ** 0`. So there's no special casing needed for `(negative number)**0`.

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/builtins.pyi:179]: Putting it near the PositiveInteger overload should be fine. Order does matter for overloads: type checkers essentially try them in order.

**Comment by AlexWaygood** [comment]
Could you add the new overload to the `pow` definitions at lines 1349 and 1383, as well?

**Review by JelleZijlstra — APPROVED** [review]

File: stdlib/builtins.pyi [hunk 2]
```diff
@@ -1383,6 +1387,8 @@ else:
     @overload
     def pow(__base: int, __exp: int, __mod: int) -> int: ...
     @overload
+    def pow(base: int, exp: Literal[0], mod: None = ...) -> Literal[1]: ...  # type: ignore[misc]
```
[the remark this answers is not in the document]
Comment (Molkree) [reply stdlib/builtins.pyi:1390]: Urgh, copypasted without noticing, thanks!

**Pull request merged**